### PR TITLE
enh(Extensions): Refacto Extension (code style) and adapt dark mode colors if needed

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
## Description

Objective 
- Refacto Extension page to match current standards
- Switch to dark mode and adapt colors if needed

 👍  Extensions Page : 
before Refacto : 
image
after Refacto : 
image

changes : 
- change the Filter to have the same Structure as in Resource Page
- add The Search Criterias Query to Search Bar and auto suggestion and auto complete 
image

-  add the same Criteria selector as in Resource Page
image

- make the [ install all ] button hidden if there is an uninstalled extension (same thing for update all button)
image

- for the Listing, Extension page keep the same design but with code refactoring and optimization

 👍 Extension details :
Extension details modal keep the same design but with code refactoring and optimization
image

 👍 Extension delete :
Extension delete modal keep the same design but with code refactoring and optimization.
image

----------------------------
Dark Mode of Extension Page

 👍 Extensions Page:
image

 👍 Extension Details:
image

 👍 Extension Delete
image

Dark mode fixes:
- the license label posistion have a white background color in dark mode

**Fixes** # (issue)

Fixed Bug 🐛 
- We can not install the extension in the extension Details modal because the install button is disabled.
- When we launche  install all and directly open extension detail it will stops the installation of the other extensions and the uninstalled extensions remain in infinite loading loop (The same for update all)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- See if the new Extension code have the same feuture as the original and in working conditions

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Extensions/Listing/ExtensionDeleteDialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/prop-types */
+/* eslint-disable react/prefer-stateless-function */
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/ExtensionDeleteDialog/index.tsx:3]:
```suggestion
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+    ) {
+      return [
+        ...getEntitiesByKeyAndVersionParam(param, equals, 'module'),
+        ...getEntitiesByKeyAndVersionParam(param, equals, 'widget'),
+      ];
+    }
+    if (modulesActive) {
+      return [...getEntitiesByKeyAndVersionParam(param, equals, 'module')];
+    }
+    // inverted because of inverse logic for switches on/off false/true
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:164]:
```suggestion
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+        action: 'update',
+        id,
+        type,
+      }),
+    })
+      .then(({ status, result }) => {
+        if (!status) {
+          showErrorMessage(result.message as string);
+        } else {
+          showSuccessMessage('successful Update');
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:202]:
```suggestion
          showSuccessMessage('Update succeeded');
```

You should use a translatedLabel here

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+      })
+      .then(({ status, result }) => {
+        if (status) {
+          setExtension(result as Extensions);
+        }
+        reloadNavigation();
+      });
+  };
+
+  const notInstallableExtensionModuleExiste = !isEmpty(
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:341]:
```suggestion
  const canInstallAllModules = !isEmpty(
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+    filter(pathEq(['version', 'installed'], false), extensions.module.entities),
+  );
+
+  const notInstallableExtensionWidgetExiste = !isEmpty(
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:345]:
```suggestion
  const canInstallAllWidgets = !isEmpty(
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 6]
```diff
@@ -0,0 +1,468 @@
...
+    filter(pathEq(['version', 'installed'], false), extensions.widget.entities),
+  );
+
+  const notUpdatableExtensionModuleExiste = !isEmpty(
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:349]:
```suggestion
  const canUpdateAllModules = !isEmpty(
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 7]
```diff
@@ -0,0 +1,468 @@
...
+    filter(pathEq(['version', 'outdated'], true), extensions.module.entities),
+  );
+
+  const notUpdatableExtensionWidgetExiste = !isEmpty(
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:353]:
```suggestion
  const canUpdateAllWidgets = !isEmpty(
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+    filter(pathEq(['version', 'outdated'], true), extensions.widget.entities),
+  );
+
+  const updatable =
+    notUpdatableExtensionModuleExiste || notUpdatableExtensionWidgetExiste;
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:358]:
```suggestion
    canInstallAllModules || canInstallAllWidgets;
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+
+  const installable =
+    notInstallableExtensionModuleExiste || notInstallableExtensionWidgetExiste;
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:361]:
```suggestion
    canUpdateAllModules || canUpdateAllWidgets;
```

**Comment by bdauria** [comment]
@Yassir-BenBOUBKER a few remarks:
- The buttons should be disabled rather than hidden
- We should rename Not installed to Uninstalled
- We need to make sure that all labels are stored in the translatedLabels.ts file (so that it's easier later on to translate them)

File: www/front_src/src/Extensions/Listing/ExtensionsHolder/index.tsx [hunk 10]
```diff
@@ -71,21 +79,23 @@ const ExtensionsHolder = ({
...
+          label: `${t(labelLicenseExpires)} ${expirationDate
+            .toISOString()
+            .slice(0, 10)}`,
         };
       }
 
       return {
         color: '#f90026',
-        label: 'License not valid',
+        label: t(labelLicenseNotValide),
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/ExtensionsHolder/index.tsx:98]:
```suggestion
        label: t(labelLicenseNotValid),
```

File: www/front_src/src/Extensions/translatedLabels.ts [hunk 11]
```diff
@@ -4,9 +4,20 @@ export const labelSearchOptions = 'Filter options';
...
+export const labelUpdateAll = 'Update all';
+export const labelAvailable = 'Available';
+export const labelDescription = 'Description';
+export const labelLastUpdate = 'Last update';
+export const labelInstall = 'Install';
+export const labelUpdate = 'Update';
+export const labelDelete = 'Delete';
+export const labelLicenseRequired = 'License required';
+export const labelLicenseExpires = 'License expires';
+export const labelLicenseNotValide = 'License not valid';
```
Comment (bdauria) [inline www/front_src/src/Extensions/translatedLabels.ts:23]:
```suggestion
export const labelLicenseNotValid = 'License not valid';
```

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**